src/content/docs/en/guides/middleware.mdx

```diff
@@ -196,6 +197,58 @@ auth response
 validation response
 ```
 
+## Rewriting
+
+<Since v="4.13.0" />
+
+The `APIContext` exposes a method called `rewrite`, which works the same way as [Astro.rewrite](/en/guides/routing/#rewrites).
+
+The `next` function optionally accept a parameter that allows to **rewrite** the current `Request`. For example, if a user isn't logged in, you could **rewrite** the current request without the need of a redirect:
+
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
+    // The user is not logged in, then we rewrite the current `Request` with a new one, and create an header
+    // The header will tell to the `/login` where the user should be sent after a successfully login.
+    return next(new Request("/login", {
+      headers: {
+        "x-redirect-to": context.url.pathname
+      }
+    }));
+  }
+
+  return next();
+};
+```
+
+The `next` function accepts the same payload of the [`Astro.redirect() function`](/en/reference/api-reference/#astrorewrite).
+
+In case you have multiple middleware functions chained via [sequence](#chaining-middleware), using `next("/")` with  a payload will rewrite the request **in place** which means that the next middleware function will receive the **new `Request`**:
+
+```js
+// This is the first middleware function that will be triggered by Astro 
+async function first(_, next) {
+  // Let's suppose the current URL is https://exmpale.com/blog
+  console.log(ctx.url.pathname) // this will log "/blog"
+  // We rewrite to a new route, the homepage 
+  return next("/")
+}
+
+// This is the second middleware function that will be triggered by Astro
+async function second(ctx, next) {
+  // `ctx` will contain new information, computed by the previous rewrite
+  console.log(ctx.url.pathname) // this will log  "/"    
+  return next()
+}
```

Comment:
```suggestion
// Current URL is https://example.com/blog

// First middleware function
async function first(_, next) {
  console.log(ctx.url.pathname) // this will log "/blog"
  // Rewrite to a new route, the homepage 
  // Return updated `ctx` which is passed to next function
  return next("/")
}

// Current URL is still https://example.com/blog

// Second middleware function
async function second(ctx, next) {
  // Receives updated `ctx`
  console.log(ctx.url.pathname) // this will log  "/"    
  return next()
}
```
Just tried to make this a bit more compact, and also emphasizing that the URL hasn't changed. Maybe something like this?